### ISSUE TITLE
Not able to add a data source for cloudwatch using ARN

### ISSUE THREAD

**Issue by santakd** [grafana/grafana#15104]
Read before posting: 

- Questions should be posted to https://community.grafana.com. Please search there and here on GitHub for similar issues before creating a new issue. 
- Checkout FAQ: https://community.grafana.com/c/howto/faq
- Checkout How to troubleshoot metric query issues: https://community.grafana.com/t/how-to-troubleshoot-metric-query-issues/50

Please include this information:
### What Grafana version are you using?
5.4.3
### What datasource are you using?
Cloudwatch
### What OS are you running grafana on?
Ubuntu 18
### What did you do?
Wanted to add a data source using ARN
### What was the expected result?
Should see the error message if the configuration was incorrect
### What happened instead?
Got an error indication but no messsage
### Include raw network request & response: get by opening Chrome Dev Tools (F12, Ctrl+Shift+I on windows, Cmd+Opt+I on Mac), go the network tab.
Nope

![image](https://user-images.githubusercontent.com/11453706/51881212-1f797780-232f-11e9-8a96-b88a138ded6e.png)

**Comment by santakd** [comment]
When clicking on the save and test, we are getting this error

![image](https://user-images.githubusercontent.com/11453706/51881716-fe198b00-2330-11e9-8147-ae97ac0c40e9.png)

**Comment by marefr** [comment]
Please check Grafana server log and include relevant information here. Could it be same problem as #14722 which have been fixed and is available in nightly build and upcoming stable release?

**Comment by santakd** [comment]
###Snippet from grafana.log
t=2019-01-29T10:18:45+0000 lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=admin error="Failed to call ec2:getAwsConfig, NoCredentialProviders: no valid providers in chain. Deprecated.\n\tFor verbose messaging see aws.Config.CredentialsChainVerboseErrors"
t=2019-01-29T10:18:45+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/tsdb/query status=500 remote_addr=108.X.Y.Ztime_ms=36 size=34 referer=http://<URL>:3000/datasources/edit/5
t=2019-01-29T10:18:52+0000 lvl=info msg="Request Completed" logger=context userId=0 orgId=0 uname= method=GET path=/ status=302 remote_addr=172.X.Y.Z time_ms=0 size=29 referer=
t=2019-01-29T10:18:53+0000 lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=admin error="NoCredentialProviders: no valid providers in chain. Deprecated.\n\tFor verbose messaging see aws.Config.CredentialsChainVerboseErrors"
t=2019-01-29T10:18:53+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/tsdb/query status=500 remote_addr=108.200.55.125 time_ms=18 size=34 referer=http://<URL>:3000/datasources/edit/5

**Comment by santakd** [comment]
Yes, it is running on AWS
It is an EC2 instance running Ubuntu

**Comment by marefr** [comment]
Maybe you need to set the environment variable `AWS_SDK_LOAD_CONFIG=1` before starting grafana so that the go sdk (library Grafana uses for connecting to Cloudwatch) gets and populates the ~/.aws/config?

I should add that I don't have that good knowledge of AWS, but maybe @xlson can shed some light on this?

**Comment by xlson** [comment]
@santakd Have you given your server an IAM role that allows it to assume the arn that you have specified?

**Comment by santakd** [comment]
Thank you @marefr and @xlson 
Let me give it a try
Trying to get the access\secret thing working from a IAM user
It was working earlier just that we have removed all IAM users and hence the ARN solution was being tried

**Comment by ahornercenteredge** [comment]
I found that I was able to assume roles in other accounts, but not able to assume a role within the same account. The IAM role assigned to the EC2 instance didn't have any impact.

**Comment by forzagreen** [comment]
My role is named `my_role`.
I had to add the the line with `"AWS": "arn:aws:iam::1234567890:role/my_role"` to my Trust Relationship. It looks then like this:


```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "",
      "Effect": "Allow",
      "Principal": {
        "AWS": "arn:aws:iam::1234567890:role/my_role",
        "Service": "ec2.amazonaws.com"
      },
      "Action": "sts:AssumeRole"
    }
  ]
}

**Comment by lucasrampoldi** [comment]
Please, verify your IAM credential in Grafana Database.

**Comment by tomsozolins** [comment]
> My role is named `my_role`. I had to add the the line with `"AWS": "arn:aws:iam::1234567890:role/my_role"` to my Trust Relationship. It looks then like this:
> 
> ```
> {
>   "Version": "2012-10-17",
>   "Statement": [
>     {
>       "Sid": "",
>       "Effect": "Allow",
>       "Principal": {
>         "AWS": "arn:aws:iam::1234567890:role/my_role",
>         "Service": "ec2.amazonaws.com"
>       },
>       "Action": "sts:AssumeRole"
>     }
>   ]
> }
> ```

How would you do this in terraform? I get Error: Self-referential block
